src/connector.rs

```diff
@@ -38,7 +38,7 @@ impl<T> HttpsConnector<T> {
 impl<T> Service<Uri> for HttpsConnector<T>
 where
     T: Service<Uri>,
-    T::Response: Connection + AsyncRead + AsyncWrite + Send + Unpin + 'static,
```

Comment:
... that is a good question, I wonder how this even compiles. [Connect requires the `Service` response to implement Connection](https://docs.rs/hyper-util/latest/hyper_util/client/legacy/connect/trait.Connect.html), which the `MaybeHttpsStream` it only does if `T` implements `Connection`... I assume in reality it would error at compile time if you were to use it outside of `TokioIo<TcpStream>`. I'll add the bound to be sure

Reply:
Ah I think I got it: Since we require `T: Service<Uri>`, `T::Response` is guaranteed to implement the traits based on the bounds in hyper-util

Reply:
Thanks! Resolving based on this thread & 6fca9a1cd5eca0de313522c244895727be38fb6d